### PR TITLE
Switch to use Serde for the Mybatis data access guide

### PR DESCRIPTION
**Pull request by timyates**

### CODE REVIEW

File: guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy [hunk 1]
```diff
@@ -28,7 +28,14 @@ class ListingArguments {
...
-    ListingArguments() {
+    private ListingArguments() {
+    }
+
+    ListingArguments(Integer offset, @Nullable Integer max, @Nullable String sort, @Nullable String order) {
+        this.offset = offset
+        this.max = max
+        this.sort = sort
+        this.order = order
     }
```
Comment (timyates) [inline guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy:39]: Serde requires the constructor

Comment (sdelamo) [reply guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy:39]: the `    private ListingArguments() {` ?

File: guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java [hunk 2]
```diff
@@ -28,7 +29,15 @@ public class ListingArguments {
     @Nullable
     private String order;
 
-    public ListingArguments() {
+    private ListingArguments() {
```
Comment (sdelamo) [inline guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java:31]: can we remove this private constructor?

File: guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java [hunk 3]
```diff
@@ -28,7 +29,15 @@ public class ListingArguments {
...
+    }
+
+    @Creator
```
Comment (sdelamo) [inline guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java]: can we remove this `@Creator`

File: guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy [hunk 4]
```diff
@@ -28,7 +28,14 @@ class ListingArguments {
     @Nullable
     private String order
 
-    ListingArguments() {
+    private ListingArguments() {
```
Comment (sdelamo) [inline guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy:31]: can we remove this private constructor

File: guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java (hunk 2 above)
Comment (timyates) [reply guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java:31]: The private constructor is used by the builder

File: guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy (hunk 1 above)
Comment (timyates) [reply guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy:39]: No, the 4 argument one

The private constructor is there for the builder

File: guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy (hunk 4 above)
Comment (timyates) [reply guides/micronaut-data-access-mybatis/groovy/src/main/groovy/example/micronaut/ListingArguments.groovy:31]: No, it's used by the builder

File: guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java (hunk 3 above)
Comment (timyates) [reply guides/micronaut-data-access-mybatis/java/src/main/java/example/micronaut/ListingArguments.java]: Gone in https://github.com/micronaut-projects/micronaut-guides/pull/1139/commits/d748c5715e9bde0468722f8ceb5e1450aafe0862

**Review by sdelamo — APPROVED** [review]

**Pull request merged**